### PR TITLE
Split ClusterResourceQuotas into different groups

### PR DESCRIPTION
**Pull request by rajivnathan**
Related Issue: https://issues.redhat.com/browse/CRT-941

Related e2e PR: https://github.com/codeready-toolchain/toolchain-e2e/pull/238

This PR splits up our ClusterResourceQuotas into different groups as a workaround for an issue where 409 conflict errors occur when creating apps via dev console. It should make 409 situations less likely to happen.

Checklist:
- [x] Manually test that the 409 errors are less likely to occur with these changes
- [x] Negligible performance difference when comparing with and without this PRs changes
- [x] Unit tests updated
- [x] e2e tests updated

### CODE REVIEW

File: deploy/templates/nstemplatetiers/advanced/cluster.yaml [hunk 1]
```diff
@@ -18,20 +18,108 @@ objects:
...
+        openshift.io/requester: ${USERNAME}
+      labels: null
+- apiVersion: quota.openshift.io/v1
+  kind: ClusterResourceQuota
+  metadata:
+    name: for-${USERNAME}-bc
+  spec:
+    quota:
+      hard:
+        count/buildconfigs.build.openshift.io: "10"
```
Comment (alexeykazakov) [inline deploy/templates/nstemplatetiers/advanced/cluster.yaml:99]: Let's maybe increase it to 30 so it matches the deployment count.
```suggestion
        count/buildconfigs.build.openshift.io: "30"
```

File: deploy/templates/nstemplatetiers/basic/cluster.yaml [hunk 2]
```diff
@@ -17,8 +17,110 @@ objects:
...
+        openshift.io/requester: ${USERNAME}
+      labels: null
+- apiVersion: quota.openshift.io/v1
+  kind: ClusterResourceQuota
+  metadata:
+    name: for-${USERNAME}-bc
+  spec:
+    quota:
+      hard:
+        count/buildconfigs.build.openshift.io: "10"
```
Comment (alexeykazakov) [inline deploy/templates/nstemplatetiers/basic/cluster.yaml:99]:
```suggestion
        count/buildconfigs.build.openshift.io: "30"
```

File: deploy/templates/nstemplatetiers/basicdeactivationdisabled/cluster.yaml [hunk 3]
```diff
@@ -17,8 +17,110 @@ objects:
...
+        openshift.io/requester: ${USERNAME}
+      labels: null
+- apiVersion: quota.openshift.io/v1
+  kind: ClusterResourceQuota
+  metadata:
+    name: for-${USERNAME}-bc
+  spec:
+    quota:
+      hard:
+        count/buildconfigs.build.openshift.io: "10"
```
Comment (alexeykazakov) [inline deploy/templates/nstemplatetiers/basicdeactivationdisabled/cluster.yaml:99]:
```suggestion
        count/buildconfigs.build.openshift.io: "30"
```

File: deploy/templates/nstemplatetiers/team/cluster.yaml [hunk 4]
```diff
@@ -18,20 +18,108 @@ objects:
...
+        openshift.io/requester: ${USERNAME}
+      labels: null
+- apiVersion: quota.openshift.io/v1
+  kind: ClusterResourceQuota
+  metadata:
+    name: for-${USERNAME}-bc
+  spec:
+    quota:
+      hard:
+        count/buildconfigs.build.openshift.io: "10"
```
Comment (alexeykazakov) [inline deploy/templates/nstemplatetiers/team/cluster.yaml:99]:
```suggestion
        count/buildconfigs.build.openshift.io: "30"
```

File: pkg/templates/nstemplatetiers/nstemplatetier_generator_whitebox_test.go [hunk 5]
```diff
@@ -526,12 +550,40 @@ func limitRangeObj(kind, cpuLimit, memoryLimit, cpuRequest, memoryRequest string
...
+func clusterResourceQuotaJobsObj() string {
+	return `{"apiVersion":"quota.openshift.io/v1","kind":"ClusterResourceQuota","metadata":{"name":"for-${USERNAME}-jobs"},"spec":{"quota":{"hard":{"count/cronjobs.batch":"30","count/daemonsets.apps":"30","count/jobs.batch":"30","count/statefulsets.apps":"30"}},"selector":{"annotations":{"openshift.io/requester":"${USERNAME}"},"labels":null}}}`
+}
+
+func clusterResourceQuotaServicesObj() string {
+	return `{"apiVersion":"quota.openshift.io/v1","kind":"ClusterResourceQuota","metadata":{"name":"for-${USERNAME}-services"},"spec":{"quota":{"hard":{"count/services":"10"}},"selector":{"annotations":{"openshift.io/requester":"${USERNAME}"},"labels":null}}}`
+}
+
+func clusterResourceQuotaBuildConfigObj() string {
+	return `{"apiVersion":"quota.openshift.io/v1","kind":"ClusterResourceQuota","metadata":{"name":"for-${USERNAME}-bc"},"spec":{"quota":{"hard":{"count/buildconfigs.build.openshift.io":"10"}},"selector":{"annotations":{"openshift.io/requester":"${USERNAME}"},"labels":null}}}`
```
Comment (alexeykazakov) [inline pkg/templates/nstemplatetiers/nstemplatetier_generator_whitebox_test.go:578]:
```suggestion
	return `{"apiVersion":"quota.openshift.io/v1","kind":"ClusterResourceQuota","metadata":{"name":"for-${USERNAME}-bc"},"spec":{"quota":{"hard":{"count/buildconfigs.build.openshift.io":"30"}},"selector":{"annotations":{"openshift.io/requester":"${USERNAME}"},"labels":null}}}`
```

**Review by alexeykazakov — APPROVED** [review]
Thanks for testing the change! Great job.
I'm approving the PR but let's increase the build config count to match the deployment count.
Also let's update the `test` tier too. It's not critical but I would keep it in sync with the basic tier for now.

**Review by xcoulon — APPROVED** [review]
Approved, pending the changes requested by Alexey with regards to the number of build configs ;)

**Review by MatousJobanek — APPROVED** [review]

**Pull request merged**